File: Algorithms/2075. Decode the Slanted Ciphertext.py
class Solution:
    def decodeCiphertext(self, encodedText: str, rows: int) -> str:
        # pass to rows col 
        cols = len(encodedText) // rows 
        arr = []
        temp = []
        for e in encodedText:
            temp.append(e)
            if len(temp) == cols:
                arr.append(temp)
                temp = []
        
        # print(arr)
        # build up str 
        output = ""
        for i in range(cols):
            # iterate directional down 
            row = 0 
            col = i 
            while row < rows and col < cols:
                output += arr[row][col]
                row += 1
                col += 1
        
        output = output.rstrip()
        return output

File: Algorithms/2075. Decode the Slanted Ciphertext.ts
function strip_str(s: string): string {
    // output: string = ""
    // front 
    let i = 0
    // for (; i < s.length; i++) {
    //     if (s[i] !== ' ') break;
    // }
    // s = s.substring(i)
    // back
    i = s.length - 1
    for (; i >= 0; i--) {
        if (s[i] !== ' ') break
    }
    s = s.substring(0, i + 1)
    // console.log(s)
    return s
}

function decodeCiphertext(encodedText: string, rows: number): string {
    let cols: number = encodedText.length / rows
    let arr: string[][] = []
    let temp: string[] = []
    for (const e of encodedText) {
        temp.push(e)
        if (temp.length === cols) {
            arr.push(temp)
            temp = []
        }
    }
    let output: string = ""
    // console.log(arr)
    for (let i = 0; i < cols; i++) {
        let row = 0
        let col = i
        while (row < rows && col < cols) {
            // console.log(arr[row][col])
            output += arr[row][col]
            row += 1
            col += 1
        }
    }
    // strip
    output = strip_str(output)
    // output = output.trim()
    return output
};